vendor/github.com/docker/distribution/registry/handlers/app.go

```diff
@@ -779,7 +779,18 @@ func (app *App) logError(context context.Context, errors errcode.Errors) {
 			errCodeKey{},
 			errMessageKey{},
 			errDetailKey{}))
-		ctxu.GetResponseLogger(c).Errorf("response completed with error")
+
+		logf := ctxu.GetResponseLogger(c).Errorf
+
+		httpStatus, ok := c.Value("http.response.status").(int)
```

Comment:
needs a comment.  looks like the goal is to log 404s for HEAD requests as Info, everything else as Error?

Assuming i'm right... shouldn't 404s for HEAD be logged as debug, not info?

Reply:
No, otherwise we will not see any attempts to make this request. Besides, it's still a mistake and we can not be 100% sure that it can be ignored.

Reply:
I assume most of these come from clients looking to see if a layer exists before pushing it?

Reply:
Yep